README.md

```diff
@@ -328,6 +287,62 @@ and it will use `data-url` attribute to define the page with comments.
 
 Also script can uses `url` property from `remark_config` object, or `window.location.href` if nothing else is defined. 
 
+## Development
+
+You can use fully functional local version to develop and test both frontend & backend.
+
+To bring it up run:
+
+```bash
+# if you mainly work on backend
+docker-compose -f compose-dev-backend.yml build
+docker-compose -f compose-dev-backend.yml up
+# if you mainly work on frontend
+docker-compose -f compose-dev-frontend.yml build
+docker-compose -f compose-dev-frontend.yml up
+```
+
+It starts Remark42 on `127.0.0.1:8080` and adds local OAuth2 provider “Dev”.
+To access UI demo page go to `127.0.0.1:8080/web`.
+By default, you would be logged in as `dev_user` which defined as admin. 
+You can tweak any of [supported parameters](#Parameters) in corresponded yml file.
+
+Backend docker compose config by default skips running frontend related tests. 
+Frontend docker compose config by default skips running backend related tests and sets `NODE_ENV=development` for frontend build. 
+
+### Backend
+
+Maybe instructions for local build here?
+
+### Frontend
+
+#### Build
+
+* install [Node.js 8](https://nodejs.org/en/) or higher;
+* install [NPM 6.1.0](https://www.npmjs.com/package/npm);
+* run `npm install` inside `./web`;
+* run `npm run build` there;
+* result files will be saved in `./web/public`.
+
+**Note** Running `npm install` will set up precommit hooks into your git repository.
+It used to reformat your frontend code using `prettier` and lint with `eslint` before every commit.
+
+#### Devserver
+
+For local development mode with Hot Reloading use `npm start` instead of `npm run build`.
+In this case `webpack` will serve files using `webpack-dev-server` on `localhost:9000`.
+By visiting `127.0.0.1:9000/web` you will get a page with main comments widget.
+communicating with demo server backend running on `https://demo.remark42.com`.
+But you will not be able to login with any oauth providers due to security reasons.
+
+You can attach to locally running backend by providing `REMARK_URL` environment variable.
+```sh
+npx cross-env REMARK_URL=http://127.0.0.1:8080 npm start
```

Comment:
I don't see any details about npx. Maybe you should remote it and leave `REMARK_URL=http://127.0.0.1:8080 npm start` only?

Reply:
`npx` is a utility that comes with npm preinstalled since npm@5.2.0.
It allows running any npm package without installing it either locally or globally.
See https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b for details.

Syntaxis that you suggest isn't supported in windows. `cross-env` is used to solve cross-platform differences.